app/services/decisions/client_decision_tree.rb

```diff
@@ -31,19 +31,19 @@ def after_has_partner
     end
 
     def after_client_details
-      if AgeCalculator.new(current_crime_application.applicant).under18?
+      if applicant.under18?
         start_address_journey(HomeAddress)
       else
         edit(:has_nino)
       end
     end
 
     def after_has_nino
-      DWP::UpdateBenefitCheckResultService.call(current_crime_application.applicant)
+      DWP::UpdateBenefitCheckResultService.call(applicant)
 
-      if current_crime_application.applicant.passporting_benefit.nil?
+      if applicant.passporting_benefit.nil?
```

Comment:
Yes I agree. I didn't want to refactor too many things outside the (already quite big) scope of this PR, and it has grown too large already.
But as there are some remaining bits and pieces hanging around regarding to edge cases with passported returned applications, this could be done separately to improve readability of the code 👍